### PR TITLE
Cleanup Twig environment config resolution.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/View/TwigView.php [hunk 1]
```diff
@@ -113,22 +113,17 @@ public function setTwig(Environment $twig): void
      */
     protected function resolveConfig(): array
     {
-        $charset = 'utf-8';
-        if (Configure::read('App.encoding') !== null) {
-            $charset = strtolower(Configure::read('App.encoding'));
```
Comment (ADmad) [inline src/View/TwigView.php]: The `strtolower()` was unnecessary. Twig's [Environment](https://github.com/twigphp/Twig/blob/3.x/src/Environment.php#L104) uses upper cased string.

File: src/View/TwigView.php [hunk 2]
```diff
@@ -113,22 +113,17 @@ public function setTwig(Environment $twig): void
...
-        }
-        $debugFlag = false;
-        if (Configure::read('App.encoding') === true) {
-            $debugFlag = true;
```
Comment (ADmad) [inline src/View/TwigView.php]: Setting debug based on `App.encoding` was just wrong.

File: src/View/TwigView.php [hunk 3]
```diff
@@ -113,22 +113,17 @@ public function setTwig(Environment $twig): void
...
+        $debug = Configure::read('debug', false);
+
+        $config = $this->readConfig() + [
+            'charset' => Configure::read('App.encoding', 'UTF-8'),
+            'debug' => $debug,
+            'cache' => $debug ? false : CACHE . 'twigView' . DS,
         ];
 
-        $config = array_replace($config, $this->readConfig());
+        if ($config['cache'] === true) {
```
Comment (ADmad) [inline src/View/TwigView.php]: The allows one to easily force cache to be used even with debug on, without having to explicitly set a path string.

File: src/View/TwigView.php [hunk 4]
```diff
@@ -113,22 +113,17 @@ public function setTwig(Environment $twig): void
...
-        }
-        $config = [
-            'cache' => CACHE . 'twigView' . DS,
-            'charset' => $charset,
-            'auto_reload' => $debugFlag,
-            'debug' => $debugFlag,
+        $debug = Configure::read('debug', false);
+
+        $config = $this->readConfig() + [
+            'charset' => Configure::read('App.encoding', 'UTF-8'),
```
Comment (othercorey) [inline src/View/TwigView.php]:
```suggestion
            'charset' => Configure::read('App.encoding', 'utf-8'),
```

match the twig docs

Comment (ADmad) [reply src/View/TwigView.php]: I always go to the source :)  https://github.com/twigphp/Twig/blob/6384ba6c3ca7695ef0aec93bdf3b55c9bb18b5de/src/Environment.php#L104

File: src/View/TwigView.php (hunk 3 above)
Comment (othercorey) [reply src/View/TwigView.php]: If users set a custom cache path, it will always cache, right? This only works with default settings?

**Pull request merged**